comp/core/secrets/component.go

```diff
@@ -15,7 +15,7 @@ import (
 // Component is the component type.
 type Component interface {
 	// Configure the executable command that is used for decoding secrets
-	Configure(command string, arguments []string, timeout, maxSize, refreshInterval int, groupExecPerm, removeLinebreak bool)
+	Configure(command string, arguments []string, timeout, maxSize, refreshInterval int, groupExecPerm, removeLinebreak bool, auditFilename string)
```

Comment:
The `Configure` function arguments are becoming to big in my opinion. Have we thought about creating a struct or something to hold the configuration data?

Reply:
I completed agree, added a Params struct for this method, it cleans up quite a bit.